### ISSUE TITLE
New syntax for generic structs missing.

### ISSUE THREAD

**Issue by souragc** [checkedc/checkedc#506]
I was playing around with the new syntax for Checked C and it seems generic structs don't have support for it yet.

```c
struct IntSet _For_any(T) {
  T *rep;
  void (*add)(T *rep, int x);
  void (*rem)(T *rep, int x);
  int (*find)(T *rep, int x);
};

struct IntSet _TyArgs(struct List) listSet;
```

The struct definition is taken from the wiki.

Currently, I get the following error:

```sh

error: expected a type argument list for a generic struct type
struct IntSet _TyArgs(struct List) listSet;
              ^
1 error generated.
```

**Comment by dtarditi** [comment]
@souragc Yes, I checked and it is not implemented yet.  In lib\Parse\ParseDeclCXX.cpp, the function [`DeclResultParser::ParseRecordTypeApplication`](https://github.com/checkedc/checkedc-clang/blob/a2843eae17d96cbfad2cf386dd60de0b849cfb6d/clang/lib/Parse/ParseDeclCXX.cpp#L2122) needs to be modified to handle the new syntax.
